components/ProgramsComponent.tsx

```diff
@@ -0,0 +1,101 @@
+"use client";
+
+import React, { useState } from 'react';
+import { Box, Typography } from '@mui/joy';
+import OutlineButton from './OutlineButton';
+import LabelledIcon from './LabelledIcon';
+
+interface ProgramsComponentProps {
+  frontTitle: string;
+  frontImageSrc: string;
+  backContent: React.ReactElement[];
+  link: string;
+  description: string;
+}
+
+
+export default function ProgramsComponent({ frontTitle, frontImageSrc, backContent, link, description }: ProgramsComponentProps) {
+	const [isFlipped, setIsFlipped] = useState(false);
+
+  const toggleComponent = () => {
+		setIsFlipped(!isFlipped);
+	};
+
+  return (
+    <Box
+      onClick={toggleComponent}
+      className={'shadow border-round'}
```

Comment:
the figma has a slightly more visible shadow when the card component is flipped as well, so if isFlipped then maybe you could set the shadow to be a bit more